src/xma/src/xmaplugin/xmaplugin.cpp

```diff
@@ -736,7 +737,7 @@ XmaCUCmdObj xma_plg_schedule_cu_cmd(XmaSession s_handle,
     
         if (!kernel_tmp1->soft_kernel && !kernel_tmp1->in_use && !kernel_tmp1->context_opened) {
 	    //Obtain lock only for a) singleton changes & b) kernel_info changes
-            std::unique_lock<std::mutex> guard1(g_xma_singleton->m_mutex);
+            std::unique_lock<std::shared_timed_mutex> guard1(g_xma_singleton->m_mutex);
```

Comment:
```suggestion
            std::unique_lock guard1(g_xma_singleton->m_mutex);
```

Reply:
template type deduction not available in c++14